pyuvdata/uvdata/mir.py

```diff
@@ -589,16 +764,39 @@ def _init_from_mir_parser(
             source_ra = np.mean(mir_data.in_data["rar"][source_mask]).astype(float)
             source_dec = np.mean(mir_data.in_data["decr"][source_mask]).astype(float)
             source_epoch = np.mean(mir_data.in_data["epoch"][source_mask]).astype(float)
-            self._add_phase_center(
-                mir_data.codes_data["source"][sou_id],
-                cat_type="sidereal",
-                cat_lon=source_ra,
-                cat_lat=source_dec,
-                cat_epoch=source_epoch,
-                cat_frame="fk5",
-                info_source="file",
-                cat_id=int(sou_id),
-            )
+            if source_epoch != 2000.0:
+                # When fed a non-J2000 coordinate, we want to convert that so that it
+                # can easily be written into CASA MS format. In this case, we take the
+                # median apparent position and translate that to ICRS
+                time_arr = Time(
+                    mir_data.in_data["mjd"][source_mask], scale="tt", format="mjd"
+                ).utc.jd
+                icrs_ra, icrs_dec = uvutils.transform_app_to_icrs(
+                    time_arr,
+                    mir_data.in_data["ara"][source_mask],
+                    mir_data.in_data["adec"][source_mask],
+                    self.telescope_location_lat_lon_alt,
+                )
+                self._add_phase_center(
+                    mir_data.codes_data["source"][sou_id],
+                    cat_type="sidereal",
+                    cat_lon=np.median(icrs_ra),
```

Comment:
For future proofing, I wonder about setting a check that `np.ptp` of the ra, dec varies less than some reasonable amount.

I think this would only matter for concatenated tracks (in which case, is that already handled when a phase center has the same name but different coordinates?).

Reply:
Fair point! I've added a check immediately after the code highlighted here, which raises a warning if a change of > 1 arcmin is seen (> 1 primary beam for SMA, which is _pretty_ uncommon for standard sources).